File: nestedfeature/noop/src/main/java/com/luigivampa92/nestedfeature/impl/NestedFeatureHolder.java
package com.luigivampa92.nestedfeature.impl;

import com.luigivampa92.nestedfeature.api.NestedFeatureApi;
import com.luigivampa92.nestedfeature.api.NestedFeatureAvailabilityChecker;
import com.luigivampa92.nestedfeature.api.NestedFeatureLauncher;

public class NestedFeatureHolder implements NestedFeatureApi {

    @Override
    public NestedFeatureAvailabilityChecker getAvailabilityChecker() {
        return new NestedFeatureAvailabilityChecker() {
            @Override
            public boolean isEnabled() {
                return false;
            }
        };
    }

    @Override
    public NestedFeatureLauncher getFeatureLauncher() {
        return new NestedFeatureLauncher() {
            @Override
            public void launch() {
            }
        };
    }
}

File: nestedfeature/impl/src/main/java/com/luigivampa92/nestedfeature/impl/NestedFeatureHolder.kt
package com.luigivampa92.nestedfeature.impl

import com.luigivampa92.nestedfeature.api.NestedFeatureApi

class NestedFeatureHolder : NestedFeatureApi {

    override fun getAvailabilityChecker() = NestedFeatureAvailabilityCheckerImpl()

    override fun getFeatureLauncher() = NestedFeatureLauncherImpl()

}